CONTRIBUTING.md

```diff
@@ -0,0 +1,48 @@
+# Introduction
+
+datafusion-ruby follows the standard Ruby gem structure. It uses [Magnus](https://github.com/matsadler/magnus) as the Ruby bindings for Rust.
+
+```
+|
+|- lib # Ruby code
+|- ext # Rust code
+```
+
+# Develop
+
+## setup
+
+```
+make install
+```
+
+## Test
+
+```
+# compile rust and run ruby tests
+make test
+```
+
+## Formating
+
+```
+make fmt
+```
+
+# FAQ
+
+## Why Magnus?
+
+As of 2022-07, there are a few popular Ruby bindings for Rust, [Rutie](https://github.com/danielpclark/rutie), [Magnus](https://github.com/matsadler/magnus) and [other alternatives](https://github.com/matsadler/magnus#alternatives). Magnus is picked because its API seems cleaner and it seems more clear about safe vs unsafe. The author of Magnus have a "maybe bias" comparison in this [reddit thread](https://www.reddit.com/r/ruby/comments/uskibb/comment/i98rds4/?utm_source=share&utm_medium=web2x&context=3). It is totally subjective and it should not be large effort if we decides to switch to different Ruby bindings fr Rust in future.
+
+## Why the module name and gem name are different?
+
+The module name `Datafusion` follows the [datafusion](https://github.com/apache/arrow-datafusion) and [datafusion-python](https://github.com/datafusion-contrib/datafusion-python). The gem name `datafusion` [is occupied in rubygems.org at 2016](https://rubygems.org/gems/datafusion), so our gem is called `arrow-datafusion`.
```

Comment:
I have a plan to use `DataFusion` for module name and `red-datafusion` for gem name.

Reply:
> red-datafusion

good name, another red-data-tools! I think of this name as well, of course, inspire by your `red-arrow`.

btw, during this PR, I have publish as `arrow-datafusion` to rubygems.org. Let me know if `datafusion-contrib` want to re-claim the ownership

Reply:
Thanks!
If you don't mind, could you add me to the owners of `arrow-datafusion` for future transfer (that may not be occurred...)?

```bash
$ gem owner -a kou arrow-datafusion
```

If you want to release more versions from https://github.com/jychen7/datafusion-ruby in the future, you don't need to add me to the owners of `arrow-datafusion`.

Reply:
OK.